### PR TITLE
TFAC-992 Starting GroupImpact (integrating into index.js)

### PR DESCRIPTION
**Pull request by jedtan**

### CODE REVIEW

File: src/utils/hooks/useOnClickOutside.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/* globals document */
+// Adopted from https://usehooks.com/useOnClickOutside/
```
Comment (jedtan) [inline src/utils/hooks/useOnClickOutside.js:2]: I lifted this from an external source and wasn't able to find a great way to test it. Let me know if this is a stopper for merge.

**Review by spicermatthews — APPROVED** [review]
LTGM.

But since we still have @kmjennison could be worth his eyes.

File: src/utils/hooks/useOnClickOutside.js (hunk 1 above)
Comment (kmjennison) [reply src/utils/hooks/useOnClickOutside.js:2]: All good

File: src/pages/index.js [hunk 2]
```diff
@@ -546,9 +560,17 @@ const Index = ({ data: fallbackData, userAgent }) => {
       setShouldShowSfacExtensionPrompt(
         !searchbarSfacPrompt && showSfacExtensionPrompt
       )
-      setShouldShowSearchbarSFACPrompt(searchbarSfacPrompt)
+      setShouldShowSearchbarSFACPrompt(
+        searchbarSfacPrompt && !(impactType === CAUSE_IMPACT_TYPES.group)
```
Comment (kmjennison) [inline src/pages/index.js:564]: Will we just never show this version of the SFAC prompt when group impact is enabled? That's probably OK temporarily, but we'll want to figure out the way forward on this. This version of the SFAC notification is not a conclusive winner on the experiment.

**Comment by kmjennison** [comment]
From a quick look over, all makes sense to me!

File: src/pages/index.js (hunk 2 above)
Comment (jedtan) [reply src/pages/index.js:564]: I'm going to file a ticket so we don't lose this context! TFAC-1217

**Pull request merged**